### PR TITLE
[11.x] WIP: SEPA Debit support

### PR DESCRIPTION
**Pull request by lsmith77**
see https://github.com/laravel/cashier/issues/435

Note when setting up the intent it is necessary to specifiy "seba_debit", f.e. if you want to allow both card and sepa_debit:

```
        $options = [
            'payment_method_types' => ['card', 'sepa_debit']
        ];

        $setupIntent = SetupIntent::create($options, Cashier::stripeOptions());
```

I managed to do a charge in testmode inside my Spark app (needed to do some modifications of course inside the Spark app too). Also noticed that there seems to be a 10k EUR limit on my account and I wanted to charge 10.8k EUR .. doh .. but with a lower amount it worked.

### CODE REVIEW

File: src/Billable.php [hunk 1]
```diff
@@ -594,9 +605,15 @@ public function updateDefaultPaymentMethodFromStripe()
...
-        if ($paymentMethod->type === 'card') {
-            $this->card_brand = $paymentMethod->card->brand;
-            $this->card_last_four = $paymentMethod->card->last4;
+        switch ($paymentMethod->type) {
+            case 'card':
+                $this->card_brand = $paymentMethod->card->brand;
+                $this->card_last_four = $paymentMethod->card->last4;
+                break;
+            case 'sepa_debit':
+                $this->card_brand = 'SEPA Debit';
```
Comment (lsmith77) [inline src/Billable.php]: following the example of `Bank Account` here .. obviously we could also store the `bank_code` here but I guess in most cases people just want to know if its a SEPA payment method.

**Comment by lsmith77** [comment]
Note when setting up the intent it is necessary to specifiy "seba_debit:

```
        $options = [
            'payment_method_types' => ['card', 'sepa_debit']
        ];

        $setupIntent = SetupIntent::create($options, Cashier::stripeOptions());
```

File: src/Billable.php [hunk 2]
```diff
@@ -430,17 +430,28 @@ public function paymentMethods($parameters = [])
             return collect();
         }
 
-        $parameters = array_merge(['limit' => 24], $parameters);
+        $parameters = array_merge(['limit' => 24, 'customer' => $this->stripe_id], $parameters);
 
         // "type" is temporarily required by Stripe...
         $paymentMethods = StripePaymentMethod::all(
-            ['customer' => $this->stripe_id, 'type' => 'card'] + $parameters,
+            ['type' => 'card'] + $parameters,
```
Comment (driesvints) [inline src/Billable.php]: Doesn't the below work?

```suggestion
            ['type' => ['card', 'sepa_debit']] + $parameters,
```

**Comment by driesvints** [comment]
Thanks for the PR. This will still need to be fleshed out a lot more with tests, an update to the `payment.blade.php` page and some updates to the `PaymentMethod` class. I might have some time this week to look at this.

File: src/Billable.php (hunk 2 above)
Comment (lsmith77) [reply src/Billable.php]: let me check once more .. IIRC I ran into issues.

**Comment by lsmith77** [comment]
I can look into doing those additional fixes sometime this week.

BTW I assume this should go into master .. or could it also go into 10.x ?
Spark is currently locked on cashier 10.x and a specific stripe API version which conflicts with master.

**Comment by driesvints** [comment]
Definitely master.

**Comment by lsmith77** [comment]
I just added some changes to `payment.blade.php` .. not tested at all.

I looked at `PaymentMethod` and did't see what should be added there.
Would you prefer if the `SEPA Debit` string we defined over there?

**Comment by lsmith77** [comment]
not sure how soon I get some more time for this .. tests are obviously also still missing.

**Comment by driesvints** [comment]
@lsmith77 think yu still need to push your changes?

File: src/Billable.php (hunk 2 above)
Comment (lsmith77) [reply src/Billable.php]: unfortunately this gives an `Invalid string: {:"0"=>"card", :"1"=>"sepa_debit"}`

**Comment by lsmith77** [comment]
oops .. tried to push to origin .. pushed

**Comment by lsmith77** [comment]
just rebased on master.

**Comment by driesvints** [comment]
@lsmith77 we made the decision to not include SEPA for now. There's still quite a bit of work with this PR and we'll need to test various scenarios regarding SCA, failed payments, delayed notifications. We want to get v11 finally out of the door now and will look again at this PR for v12 maybe. Thanks for your work. Don't consider this lost, I'll re-look at this once I get to it.

**Comment by lsmith77** [comment]
ok. no worries. if I bring this to prod on my side I can do this in a fork. just wondering how do you plan to retain the work for v12? I guess a closed PR will remain available and its linked in the issue that remains open.

**Comment by driesvints** [comment]
@lsmith77 yeah the code is still here and the issue is linked 👍

**Pull request closed without merging**